tsuru/client/tag.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 tsuru-client authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/tsuru/tsuru/cmd"
+	"github.com/tsuru/tsuru/service"
+)
+
+type TagList struct{}
+
+type tag struct {
+	Name             string
+	Apps             []string
+	ServiceInstances map[string][]string
+}
+
+func (t *TagList) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:  "tag-list",
+		Usage: "tag-list",
+		Desc:  `Retrieves and shows a list of tags with the respective apps and service instances.`,
+	}
+}
+
+func (t *TagList) Run(context *cmd.Context, client *cmd.Client) error {
+	apps, err := loadApps(client)
+	if err != nil {
+		return err
+	}
+	services, err := loadServices(client)
+	if err != nil {
+		return err
+	}
+	return t.Show(apps, services, context)
+}
+
+func (t *TagList) Show(apps []app, services []service.ServiceModel, context *cmd.Context) error {
+	tagList := processTags(apps, services)
+	if len(tagList) > 0 {
```

Comment:
Nitpick, but could this be `if len(tagList) == 0 { return nil }`. Just to avoid adding indentation levels.